core/Access.php

```diff
@@ -138,19 +143,15 @@ public function __construct()
      */
     public function reloadAccess(Auth $auth = null)
     {
-        if (!is_null($auth)) {
-            $this->auth = $auth;
-        }
+        $this->resetSites();
 
-        // if the Auth wasn't set, we may be in the special case of setSuperUser(), otherwise we fail
-        if (is_null($this->auth)) {
```

Comment:
This `if` is not need as we would check for `hasSuperUserAccess` afterwards anyway